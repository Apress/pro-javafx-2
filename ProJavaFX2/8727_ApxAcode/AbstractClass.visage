/*
 * @test
 * @run
 */

// The interface
abstract class Food {
  abstract function getName():String;
  abstract function whatsInIt():String;
}

// The implementation
class Hamburger extends Food {
  override function getName() {
    "Hamburger"
  }
  override function whatsInIt() {
    "beef patties"
  }
}

// A function that gives the calling code an instance of Food
function getTodaysSpecial():Food {
  Hamburger {}
}

// The calling code, no direct mentioning of Hamburger is made here
var food = getTodaysSpecial();
println("Today's special is {food.getName()}.  "
        "It has {food.whatsInIt()} in it.");
