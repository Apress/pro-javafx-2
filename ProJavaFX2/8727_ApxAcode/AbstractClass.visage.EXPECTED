Today's special is Hamburger.  It has beef patties in it.
